test/e2e/traffic/http.go

```diff
@@ -150,6 +151,11 @@ func (hs *HTTPServer) Start(ctx context.Context, ns *network.NetNS) error {
 			w.Header().Set("Content-Type", "application/octet-stream")
 			fileSize := hs.cfg.ChunkCount * hs.cfg.ChunkSize
 			w.Header().Set("Content-Length", strconv.Itoa(fileSize))
+			if hs.cfg.ExpectedAddr != "" {
+				if  hs.cfg.ExpectedAddr == r.RemoteAddr {
```

Comment:
maybe it makes sense to record any remote addr and then compare it later w/o storing `ExpectedAddr` in the config?

Reply:
Probably yes, fixed